src/main/java/com/atlassian/bitbucket/jenkins/internal/jenkins/oauth/servlet/AuthorizeAction.java

```diff
@@ -0,0 +1,196 @@
+package com.atlassian.bitbucket.jenkins.internal.jenkins.oauth.servlet;
+
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.Randomizer;
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.exception.InvalidTokenException;
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.token.ServiceProviderToken;
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.token.ServiceProviderTokenStore;
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.util.OAuthProblemUtils;
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Action;
+import hudson.model.Descriptor;
+import jenkins.model.Jenkins;
+import net.oauth.OAuthMessage;
+import net.oauth.OAuthProblemException;
+import net.oauth.server.OAuthServlet;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.HttpResponses;
+import org.kohsuke.stapler.StaplerRequest;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.security.Principal;
+import java.time.Clock;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import static com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.token.ServiceProviderToken.Authorization;
+import static javax.servlet.http.HttpServletResponse.SC_UNAUTHORIZED;
+import static jenkins.model.Jenkins.ANONYMOUS;
+import static net.oauth.OAuth.*;
+import static net.oauth.OAuth.Problems.*;
+
+public class AuthorizeAction extends AbstractDescribableImpl<AuthorizeAction> implements Action {
```

Comment:
This class was named "AuthorizeAction" to try to fit in line with existing Jenkins naming conventions, but these are not being observed elsewhere for actions- we seem to be using the terms "Configuration" and "Endpoint". If we want to keep using those conventions, this one should probably be renamed "ConsumerTokenConfiguration" or something similar.

Reply:
`Action` was used in the general sense and not in Jenkins sense. This is when user authorizes. I can rename it to `AuthorizeConfirmationPage` if that is more clear. What do you think?

Reply:
The `Action` label appears fairly widely (though also at times inconsistently) in the Jenkins code base, so if we're not using it, I think we should avoid it altogether. I'm happy with `AuthorizeConfirmationPage` (or maybe `AuthorizeConfirmationForm`, which we could apply to any page with a form component?)